services/wfm/src/main/java/org/openkilda/wfm/topology/floodlightrouter/bolts/DiscoveryBolt.java

```diff
@@ -0,0 +1,162 @@
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.floodlightrouter.bolts;
+
+import static org.openkilda.messaging.Utils.MAPPER;
+
+import org.openkilda.messaging.Message;
+import org.openkilda.wfm.topology.AbstractTopology;
+import org.openkilda.wfm.topology.floodlightrouter.ComponentType;
+import org.openkilda.wfm.topology.floodlightrouter.Stream;
+import org.openkilda.wfm.topology.floodlightrouter.service.FloodlightTracker;
+import org.openkilda.wfm.topology.floodlightrouter.service.MessageSender;
+import org.openkilda.wfm.topology.floodlightrouter.service.RouterService;
+import org.openkilda.wfm.topology.utils.AbstractTickStatefulBolt;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.storm.state.InMemoryKeyValueState;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Set;
+
+
+public class DiscoveryBolt extends AbstractTickStatefulBolt<InMemoryKeyValueState<String, RouterService>>
+        implements MessageSender {
+    private static final Logger logger = LoggerFactory.getLogger(DiscoveryBolt.class);
+    private static final String ROUTER_SERVICE = "ROUTER_SERVICE";
+
+
+    private final Set<String> floodlights;
+    private final long floodlightAliveTimeout;
+    private final long floodlightAliveInterval;
+
+    private transient RouterService routerService;
+
+    protected OutputCollector outputCollector;
+
+    private Tuple currentTuple;
+
+    public DiscoveryBolt(Set<String> floodlights, long floodlightAliveTimeout, long floodlightAliveInterval) {
+        this.floodlights = floodlights;
+        this.floodlightAliveTimeout = floodlightAliveTimeout;
+        this.floodlightAliveInterval = floodlightAliveInterval;
+    }
+
+    @Override
+    protected void doTick(Tuple tuple) {
+        currentTuple = tuple;
+        routerService.doPeriodicProcessing(this);
+    }
+
+    @Override
+    protected void doWork(Tuple input) {
+        String sourceComponent = input.getSourceComponent();
+        currentTuple = input;
+        Message message = null;
+        try {
+            String json = input.getValueByField(AbstractTopology.MESSAGE_FIELD).toString();
+            message = MAPPER.readValue(json, Message.class);
```

Comment:
Why not use the spout that deserializes message and you don't have to do it by yourself.

Reply:
will change it in follow up patch